sway_libs/src/ownership/ownable.sw

```diff
@@ -0,0 +1,52 @@
+library ownable;
+
+dep errors;
+dep events;
+dep ownable_storage;
+
+use errors::AccessError;
+use events::{OwnershipRenounced, OwnershipSet, OwnershipTransferred};
+use ownable_storage::OWNER;
+use std::{auth::msg_sender, hash::sha256, logging::log, storage::{get, store}};
+
+#[storage(read)]
+pub fn only_owner() {
+    let owner = get::<Option<Identity>>(OWNER);
+    require(owner.is_some() && msg_sender().unwrap() == owner.unwrap(), AccessError::NotOwner);
+}
+
+#[storage(read)]
+pub fn owner() -> Option<Identity> {
+    get::<Option<Identity>>(OWNER)
+}
+
+#[storage(read, write)]
+pub fn renounce_ownership() {
```

Comment:
I don't see the purpose of this function.  If an owner renounces ownership of a contract then anyone can call the `set_ownership` function and gain restricted access to the contract in question.  IMO `renounce_ownership` should either set the owner to the 0 address and render all restricted access to the contract useless (if we do this we should add warnings to the code and docs).  Or `renounce_ownership` should be removed and restricted access can only be changed by using the `transfer_ownership` function

Reply:
This is definitely something I thought of when writing this. I would like to keep `renounce_ownership` as it follows openzeppilin's implementation of ownership and is something that Solidity developers are familiar with. 

Since this function should be called in the developer's constructor, in practice it shouldn't ever be called again... But it does leave the door open for them to shoot themselves in the foot.

I would also like to avoid the use of the 0 address as calling `owner()` should return `None`. Adding an enum to ensure that the `set_ownership()` function cannot be called twice is another approach but will incur more gas costs.

Would like to hear input from others as well on this

Reply:
Non-initialized ownership and burned ownership are two distinct states - anyone using this library should be forced to handle them explicitly and separately. 

Maybe ownership could be an `enum` instead of an `option`, containing 3 variants for uninitialized, set, and burned ownership. We could provide some helper functions to `unwrap` the various cases.

Reply:
I agree with @simonr0204 that we should have 3 variants for uninitialized, set, and renounced(burned) ownership.  Looking at the openzeppelin docs their implementation of `renounce_ownership` makes it so no one can set ownership after the function is called.  We should also have this behavior.

Reply:
I've updated the library accordingly